packages/default-theme/components/modals/SwLoginModal.vue

```diff
@@ -1,53 +1,60 @@
 <template>
-  <SfModal
-    :visible="isOpen"
-    @close="closeHandler()"
-    transitionOverlay="fade"
-    transitionModal="fade"
-    >
-    <div> Login into your account </div>
-    <SfAlert
-      v-if="error"
-      type="danger"
-      :message="error"
-    />
-    <SfInput
-      v-model="login"
-      label="Login"
-      :valid="!$v.login.$invalid"
-      error-message="Login is required."
-      :disabled="isLoading">
-      <template #errorMessage="{ errorMessage }">
-        <SfIcon icon="info_shield" size="10px" color="#E22326" style="margin-right: 4px; display: inline-block"/> {{errorMessage}}
-      </template>
-    </SfInput>
-    <SfInput
-      v-model="password"
-      label="Password"
-      :valid="!$v.password.$invalid"
-      error-message="Password is required."
-      :disabled="isLoading"
-      type="password">
-      <template #errorMessage="{ errorMessage }">
-        <SfIcon icon="info_shield" size="10px" color="#E22326" style="margin-right: 4px; display: inline-block"/> {{errorMessage}}
-      </template>
-    </SfInput>
-    <SfButton :disabled="isLoading || $v.$invalid" @click="invokeLogin"> 
-      Login 
-    </SfButton>
-    <SfLoader :loading="isLoading" />
-  </SfModal>
+  <div id="sw-login-modal">
+    <SfModal :visible="isOpen" @close="closeHandler">
+      <transition name="fade" mode="out-in">
+        <div class="sw-login-modal__wrapper">
+          <component :is="component" :key="key" @success="$emit('close')" />
+          <div v-if="component !== 'SwResetPassword'" class="action">
+            <SfButton
+              class="sf-button--text button--muted"
+              @click="component = 'SwResetPassword'"
+            >
+              Forgotten password?
+            </SfButton>
+          </div>
+
+          <div class="bottom">
+            <template v-if="component !== 'SwRegister'">
+              Don't have and account yet?
+              <SfButton
+                class="sf-button--text"
+                @click="component = 'SwRegister'"
+              >
+                Register today?
+              </SfButton>
+            </template>
+          </div>
+          <div v-if="component !== 'SwLogin'" class="action">
+            <SfButton
+              class="sf-button--text button--muted"
+              @click="component = 'SwLogin'"
+            >
+              or try to log in again.
+            </SfButton>
+          </div>
+        </div>
+      </transition>
+    </SfModal>
+  </div>
 </template>
-<script>
 
-import { SfIcon, SfModal, SfInput, SfButton, SfLoader, SfAlert } from '@storefront-ui/vue'
-import { VuelidateMixin } from '@vuelidate/core'
-import { required } from '@vuelidate/validators'
+<script>
+import { SfButton, SfModal, SfAlert } from '@storefront-ui/vue'
 import { useUser } from '@shopware-pwa/composables'
+import SwRegister from '../SwRegister'
```

Comment:
Login is default, but other components should be loaded on demand
```suggestion
const SwRegister = () => import("./SwRegister");
```